cassiopeia/plotting/local.py

```diff
@@ -0,0 +1,829 @@
+import warnings
+from collections import defaultdict
+from typing import Dict, List, Optional, Tuple, Union
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+from matplotlib.colors import hsv_to_rgb
+from typing_extensions import Literal
+from tqdm import tqdm
+
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.mixins import PlottingError, PlottingWarning
+from cassiopeia.plotting import utilities
+from cassiopeia.preprocess import utilities as preprocess_utilities
+
+
+def compute_colorstrip_size(
+    node_coords: Dict[str, Tuple[float, float]],
+    anchor_coords: Dict[str, Tuple[float, float]],
+    loc: Literal["left", "right", "up", "down", "polar"],
+) -> float:
+    """Compute the size of colorstrip boxes.
+
+    This function computes two sizes: the width and height of the colorstrip
+    boxes. The height is selected to be the value such that the boxes are
+    tightly arranged next to one another. The width is selected to be 5% of the
+    tree depth.
+
+    Args:
+        node_coords: Node coordinates.
+        anchor_coords: Anchor coordinates.
+        loc: The location of the box relative to the anchor.
+
+    Returns:
+        Two floats representing the width and height of the box.
+    """
+    min_pos = np.inf
+    max_pos = -np.inf
+    min_depth = np.inf
+    max_depth = -np.inf
+    for x, y in node_coords.values():
+        pos, depth = x, y
+        if loc in ("left", "right"):
+            pos, depth = y, x
+        min_pos = min(min_pos, pos)
+        max_pos = max(max_pos, pos)
+        min_depth = min(min_depth, depth)
+        max_depth = max(max_depth, depth)
+    return (
+        (max_depth - min_depth) * 0.05,
+        (max_pos - min_pos + 1) / len(anchor_coords),
+    )
+
+
+def create_categorical_colorstrip(
```

Comment:
Do you think it would be worthwhile to support users passing in their own `value_mapping` dictionaries?